### PR TITLE
Refactor Identities API

### PR DESCRIPTION
**Pull request by Hayden-IO**
Instead of providing raw strings that are up to the client to determine how to parse, return typed structs. Encoded keys are included and should be parsed based on the returned type.

### CODE REVIEW

**Comment by Hayden-IO** [comment]
FYI: TUF requires a new root because go-tuf dropped support for hex-encoded keys, as they were not conformant to the TUF spec.

File: pkg/pki/tuf/tuf.go [hunk 1]
```diff
@@ -119,6 +123,7 @@ func NewPublicKey(r io.Reader) (*PublicKey, error) {
 	db := verify.NewDB()
 	for id, k := range root.Keys {
 		if err := db.AddKey(id, k); err != nil {
+			fmt.Println("potato")
```
Comment (bobcallaway) [inline pkg/pki/tuf/tuf.go]: left in by mistake?

**Review by bobcallaway — DISMISSED** [review]
lgtm minus one errant 🥔 

also, since this changes a public API surface we should probably bump into a new minor release.

File: pkg/pki/tuf/tuf.go (hunk 1 above)
Comment (Hayden-IO) [reply pkg/pki/tuf/tuf.go]: I was hungry

**Comment by Hayden-IO** [comment]
@bobcallaway Agreed on the minor bump.

**Review by bobcallaway — APPROVED** [review]

**Pull request merged**